File: src/leetcode/python/BinaryTreeInorderTraversal_94.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class BinaryTreeInorderTraversal_94(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []

        def add_val(node):
            if not node:
                return
            add_val(node.left)
            if node.val != '#':
                res.append(node.val)
            add_val(node.right)

        add_val(root)
        return res

File: src/leetcode/java/BinaryTreeInorderTraversal_94.java
package leetcode.java;

import java.util.ArrayList;
import java.util.List;

public class BinaryTreeInorderTraversal_94 {
  public List<Integer> inorderTraversal(TreeNode root) {
    List<Integer> list = new ArrayList<Integer>();
    addToList(root, list);
    return list;
  }

  public static void addToList(TreeNode root, List<Integer> list) {
    if (root == null) return;
    addToList(root.left, list);
    if (root.val != '#') list.add(root.val);
    addToList(root.right, list);
  }

  public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
  }
}
